File: leetcode/53.maximum-subarray.py
# @leet start
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        best = nums[-1]
        curr = 0

        for val in nums:
            curr = max(curr, 0)
            curr += val
            best = max(best, curr)

        return best


# @leet end

File: leetcode/53.maximum-subarray.ts
// @leet start
function maxSubArray(nums: number[]): number {
  let best = nums[0];
  let curr = 0;

  for (const num of nums) {
    curr = curr < 0 ? num : curr + num;
    best = Math.max(best, curr);
  }

  return best;
}
// @leet end
